src/components/GalleryPage/Gallery.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { graphql, useStaticQuery } from 'gatsby';
+import Img from 'gatsby-image';
+import Container from '@material-ui/core/Container';
+import Slider from 'react-slick';
+import 'slick-carousel/slick/slick.css';
+import 'slick-carousel/slick/slick-theme.css';
+import './gallery.css';
+
+export default function Gallery() {
+	const settings = {
+		dots: true,
+		infinite: true,
+		speed: 500,
+		slidesToShow: 1,
+		slidesToScroll: 1,
+		className: 'slides'
+	};
+
+	const data = useStaticQuery(graphql`
+		{
+			carouselImages: allFile(filter: {relativePath: {glob:"carousel/*"}}) {
+				nodes {
+					id
+					relativePath
+					childImageSharp {
+						fluid {
+							...GatsbyImageSharpFluid
+						}
+					}
+				}
+			}
+		}
+	`);
+
+	const image= []
+
+	for (let i =0; i<=data.carouselImages.nodes;i++) {
```

Comment:
1. The index should be compared against the length of the array, rather than the array itself.

2. Because JavaScript is 0-indexed, we should use `<` to compare the index and the length rather than `<=`.

```suggestion
	for (let i = 0; i < data.carouselImages.nodes.length; i++) {
```